src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -193,6 +193,16 @@
     ],
 )
 
+snowflake_dialect.sets("refreshmode_types").clear()
+snowflake_dialect.sets("refreshmode_types").update(
+    ["AUTO", "FULL", "INCREMENTAL"],
+)
+
+snowflake_dialect.sets("initialize_types").clear()
+snowflake_dialect.sets("initialize_types").update(
+    ["ON_CREATE", "ON_SCHEDULE"],
+)
```

Comment:
Does this need to be a set rather than just including directly in the grammar?

Reply:
Well initially I actually have tried to include it directly as you suggested, but then my test cases start failing - therefore I had to do some additional grammar modifications on the segment at question. Even with that it then started giving me false positives on the TCs (e.g the tests were passing even if my value segment of REFRESH_MODE had incorrect value).

Nevertheless, I think we should at least stick with this kind of separation between the dialect values and the segment definition themself (maybe even think to move it to different file ?)